src/mail/editor/MailEditor.ts

```diff
@@ -159,59 +174,72 @@ export class MailEditor implements Component<MailEditorAttrs> {
 
 		// call this async because the editor is not initialized before this mail editor dialog is shown
 		this.editor.initialized.promise.then(() => {
-			this.editor.setHTML(model.getBody())
-			// Add mutation observer to remove attachments when corresponding DOM element is removed
-			new MutationObserver(onEditorChanged).observe(this.editor.getDOM(), {
-				attributes: false,
-				childList: true,
-				subtree: true,
-			})
-			// since the editor is the source for the body text, the model won't know if the body has changed unless we tell it
-			this.editor.addChangeListener(() => model.setBody(replaceInlineImagesWithCids(this.editor.getDOM()).innerHTML))
-			this.editor.addEventListener("pasteImage", ({ detail }: ImagePasteEvent) => {
-				const items = Array.from(detail.clipboardData.items)
-				const imageItems = items.filter((item) => /image/.test(item.type))
-				if (!imageItems.length) {
-					return false
-				}
-				const file = imageItems[0]?.getAsFile()
-				if (file == null) {
-					return false
-				}
-				const reader = new FileReader()
-				reader.onload = () => {
-					if (reader.result == null || "string" === typeof reader.result) {
-						return
+			model.getExternalImageRule().then((externalImageRule) => {
+				this.editor.setHTML(model.getBody())
+
+				if (externalImageRule === ExternalImageRule.Block) {
+					this.blockExternalContent = true
+					let blockingStatus = ContentBlockingStatus.Block
+
+					if (externalImageRule === ExternalImageRule.Block) {
+						this.alwaysBlockExternalContent = true
+						blockingStatus = ContentBlockingStatus.AlwaysBlock
 					}
-					const newInlineImages = [createDataFile(file.name, file.type, new Uint8Array(reader.result))]
-					model.attachFiles(newInlineImages)
-					this.insertInlineImages(model, newInlineImages)
+
+					this.updateExternalContentStatus(blockingStatus)
+				} else if (externalImageRule === ExternalImageRule.Allow && model.getPreviousMail()?.authStatus === MailAuthenticationStatus.AUTHENTICATED) {
+					this.blockExternalContent = false
+					this.updateExternalContentStatus(ContentBlockingStatus.AlwaysShow)
+				} else if (externalImageRule === ExternalImageRule.None && model.isUserPreviousSender()) {
+					this.blockExternalContent = true
+					this.updateExternalContentStatus(ContentBlockingStatus.Block)
+				} else {
+					this.blockExternalContent = a.doBlockExternalContent()
+
+					// As we don't have a defined rule we lead the decision
+					// to show external content to the user and just process the inline images
+					this.processInlineImages()
```

Comment:
This chunk is kinda weird.

1. We do set `blockExternalContent` in most branches and then we call `updateExternalContentStatus()` which *also* sets it?
2. We do `processInlineImages()` in all cases but we don't see it, because it's hidden inside  `updateExternalContentStatus()`

It seems like you shouldn't be calling `updateExternalContentStatus()` in here?

Also I do not understand what `alwaysBlockExternalContent` is for

Reply:
This ```alwaysBlockExternalContent``` is used when we have a explicit rule to block images, so we don't even show the banner for the user.

Reply:
i'd put it as a comment in the code